src/ioctls/system.rs

```diff
@@ -374,6 +376,46 @@ impl Kvm {
         }
     }
 
+    /// AArch64 specific function to create a VM fd using the KVM fd with flexible IPA size.
+    ///
+    /// See the arm64 section of KVM documentation for `KVM_CREATE_VM`.
+    /// A call to this function will also initialize the size of the vcpu mmap area using the
+    /// `KVM_GET_VCPU_MMAP_SIZE` ioctl.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use kvm_ioctls::{Kvm, Cap};
+    /// let kvm = Kvm::new().unwrap();
+    /// // Check if the ArmVmIPASize cap is supported.
+    /// if kvm.check_extension(Cap::ArmVmIPASize) {
+    ///     let vm = kvm.create_vm_with_ipa_size(40).unwrap();
+    ///     // Check that the VM mmap size is the same reported by `KVM_GET_VCPU_MMAP_SIZE`.
+    ///     assert!(vm.run_size() == kvm.get_vcpu_mmap_size().unwrap());
+    /// }
+    /// ```
+    ///
+    #[cfg(any(target_arch = "aarch64"))]
+    pub fn create_vm_with_ipa_size(&self, ipa_size: u32) -> Result<VmFd> {
```

Comment:
nit: Besides the ioctl call, these function looks pretty much like `create_vm`. It would be nice to not duplicate the code if possible. Since they're only two functions for now, I wouldn't stress to much on it though.

Reply:
This is actually my original thought. However I was a little bit worried that this might be a breaking change. If we add another parameter called `ipa_size` to `create_vm`, it might break anything which used `create_vm` function. Although we probably can give it a default parameter, this is still not ideal...Therefore I just used a safer way, and anyone need to create a VM with flexible IPA size can call `create_vm_with_ipa_size` instead of `create_vm`.